akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -337,7 +337,7 @@ abstract class SupervisorStrategy {
       }
       decision match {
         case Resume   ⇒ publish(context, Warning(child.path.toString, getClass, logMessage))
-        case Escalate ⇒ // don't log here
+        case Escalate if (!cause.isInstanceOf[AssertionError]) ⇒ // don't log here
```

Comment:
Hmm, I'm not sure this is the right place to add this logging. If this supervisor is escalating the exception, `handleFailure` above returns `false` and the actor should re-throw the exception. Shouldn't it be the responsibility of the parent supervisor to correctly handle (and perhaps log) the exception?